impeller/golden_tests/golden_digest.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "flutter/impeller/golden_tests/working_directory.h"
+
+namespace impeller {
+namespace testing {
+
+/// Manages a global variable for tracking instances of golden images.
+class GoldenDigest {
+ public:
+  static GoldenDigest* Instance();
+
+  void AddImage(const std::string& test_name,
+                const std::string& filename,
+                int32_t width,
+                int32_t height);
+
+  /// Writes a "digest.json" file to `working_directory`.
+  ///
+  /// Returns `true` on success.
+  bool Write(WorkingDirectory* working_directory);
+
+ private:
+  FML_DISALLOW_COPY_AND_ASSIGN(GoldenDigest);
+  GoldenDigest();
+  struct Entry {
+    std::string test_name;
+    std::string filename;
+    int32_t width;
+    int32_t height;
```

Comment:
No action needed, but why isn't this just included in the file itself? Or is this for a human verifying to make sure something didn't go wrong if the file is suddenly smaller than expected or something?

Reply:
Storing it in the image file would require a library to parse the image to get the width and height in the harvester.  Since we already know it here, it was more convenient to report it.  Our dart instances in the engine have a limited set of packages available so they might not even have the ability to easily parse image headers.